src/Microsoft.Data.SqlClient/netfx/src/Microsoft/Data/Interop/SNINativeManagedWrapperX64.cs

```diff
@@ -89,7 +89,7 @@ internal static class SNINativeManagedWrapperX64
         internal static extern uint SNIGetInfoWrapper([In] SNIHandle pConn, SNINativeMethodWrapper.QTypes QType, out ProviderEnum provNum);
 
         [DllImport(SNI, CallingConvention = CallingConvention.Cdecl, EntryPoint = "SNIInitialize")]
-        internal static extern uint SNIInitialize([In] IntPtr pmo);
+        internal static extern uint SNIInitialize([In] bool useSystemDefaultSecureProtocols, [In] IntPtr pmo);
```

Comment:
This is a partial revert of the changes in #1824. The native SNI interface here still has 2 parameters and I don't know why things work in x86/x64 with only 1. Without this revert, I get an assert when running in ARM64 mode because pmo is not what the native SNIInitialize function expects. Oddly, I _don't_ get an assert when running the exact same code in x86 or x64 mode, which explains why the change passed CI, but not why it doesn't assert.

Reply:
I believe @DavoudEshtehari found it odd that the SNI revert for the `useSystemDefaultSecureProtocols` was passing on the pipeline before the new revert SNI changes was available as it won't be available until the next SNI preview release.

Reply:
We can create a new PR with the sni version bump to reapply the removal of `useSystemDefaultSecureProtocols`.

Reply:
To explain the reason things "work" in x86 and x64 tests, the pmo parameter has a default value of NULL in the SNIInitialize method so there was no interface issue. Since we hard code pmo to 0 in MDS, the previous change just ended up sending false/0 for useSystemDefaultSecureProtocols and pmo became NULL. In SNI, pmo has an assert(pmo == 0) on it but asserts are suppressed in release x64/x86 builds and pmo doesn't actually do anything in our SNI code paths (we could remove it completely) so there was no negative effect. The SNI arm release builds were missing the NDEBUG setting so asserts were firing and popping up windows.